os/fs/driver/mtd/smart.c

```diff
@@ -1774,12 +1774,6 @@ static int smart_scan(FAR struct smart_struct_s *dev)
 	totalsectors = dev->totalsectors;
 
 	dev->reservedsector = SMART_FIRST_ALLOC_SECTOR;
-#ifdef CONFIG_SMARTFS_JOURNALING
```

Comment:
um...many things in this file should be removed.
1. HEADER_IS_CLEAN...
2. header & contents checking in smart_scan(You had better compare with first version of smart.c some logic seems that deleted regarding signature..)
   -> If this is removed, you can remoive smart_validate_crc for nxfuse.
3. Verify Erase logic..

Reply:
Hi, 
- In step 2, you refer to the obsolete loop that used to check all sectors of every possible size in smartfs to find the Smartfs header?
If yes then that is already removed.
You wish to remove smart_validate_sector form smart_scan? Then smart_scan will not check the authenticity of sectors, new journaling part will take care of it?

- Step 1, If first smart_validate_sector() is removed from smart_scan part, only then HEADER_IS_CLEAN can be removed.

3. Could you elaborate on what is to be done in Step 3 "Verify Erase Logic"?

Reply:
Function smart_validate_crc() is a different case,
I don't think it can be removed from mtd/smart.c so easily.
It is used by smart_writesector and smart_readsector also in addition to smart_validate_sector.
I think you were referring to smart_validate_sector only in your comment above.

Reply:
CRC will be checked in journal recovery, so we don't have to keep values like a 
SMART_ERASEDSTATE_CRC, so we don't have to keep 2 of smart_validate_crc.

Reply:
header & contents checking means 'smart_validate_erase_block'
As I wrote new journal design, scan will create mapping table and adjust release & free count.
Recovery will handle everything.

Reply:
The CRC checking part of smart_scan() function can be removed so smart_validate_crc() isn't required here.
However, smart_validate_crc() is required in smart_writesector/readsector, so we should remove the CRC check in writesector/readsector also?
Only then we can remove SMART_ERASEDSTATE_CRC.